src/tickit/core/typedefs.py

```diff
@@ -76,6 +76,13 @@ class Input:
     changes: Changes
 
 
+@dataclass(frozen=True, init=True)
+class StopComponent:
+    """An immutable data container for a Component shutdown flag."""
+
+    flag: bool = True
```

Comment:
I think we may have started talking about this earlier but never finished: It's perfectly okay to leave this class blank and just put `...`. It's still good software engineering because you're anticipating having to put data in there someday and reducing the amount of refactoring you need to do when that day comes.